src/openqa_client/client.py

```diff
@@ -377,3 +377,34 @@ def get_jobs(
             # there as it only considers the jobs queried.
             jobdicts = self.find_clones(jobdicts)
         return jobdicts
+
+    def get_latest_build(
+        self,
+        group_id: int,
+        all_passed: bool = True,
+        sorted_key: Callable = float,
+        default_return: str = "0",
+    ) -> str:
+        """Identify latest build number in target Job Group
+
+        Args:
+            group_id (int): Job Group ID
+            all_passed (bool, optional): Controls whether just last build will be selected
+                or the last with all jobs passed. Defaults to True.
+            sorted_key (Callable, optional): To find the latest build we need to order the builds.
+                The specified callback will be used in the sorted function to sort builds.
+                Defaults to float.
+            default_return (str, optional): In case all available builds have no acceptable format
+                or there are no builds in this job group this value specifies what to return.
+                Defaults to "0".
+
+        Returns:
+            str: string representation of last BUILD
+        """
+        resp = self.openqa_request("GET", f"/group_overview/{group_id}.json")
+        if all_passed:
+            passed = [r for r in resp["build_results"] if r["all_passed"] == 1]
+            builds = [r["build"] for r in passed if r["build"].isdigit()]
+        else:
+            builds = [r["build"] for r in resp["build_results"] if r["build"].isdigit()]
```

Comment:
the `isdigit()` filter here seems a bit arguable. For a start, it clashes with the use of `float` as the default sort key. If we're only going to accept integer builds, why not use `int` as the sort key? (note that `"123.456".isdigit()` is False). also, if we allow custom sort keys, why only allow integer builds? a custom sort key could potentially meaningfully sort non-integer builds. and what possible custom sort key would you want to use for a list of (only) integer strings that wouldn't give the same result as `float` or `int`?

I'd say consider:

1. Either make the default sort key `int` or change this `isdigit()` check to allow floating-point numbers
2. Don't apply the `isdigit()` filter if a custom sort key is specified (and if the custom sort key blows up on the build values, well, that's a problem for the caller, I think)

Reply:
done